Bottom: 3236f76bfa2e0d92d2afc03006f1c7d0929c8c38
Top:    f600af05289e413bef6b724a0bc9235a09d6aa47
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-16 13:02:18 +0800

Refresh of autofs-5.0.7-setup-program-map-env-from-macro-table.patch

---

diff --git a/lib/macros.c b/lib/macros.c
index 90184bf..33c2ada 100644
--- a/lib/macros.c
+++ b/lib/macros.c
@@ -437,6 +437,9 @@ void macro_setenv(const struct substvar *table)
 		sv = sv->next;
 	}
 
+	error(LOGOPT_ANY, "table %p", table);
+	dump_table(table);
+
 	/* Next set environment from the local table */
 	while (lv) {
 		if (lv->def)
diff --git a/modules/lookup_program.c b/modules/lookup_program.c
index b2057fe..7e22b38 100644
--- a/modules/lookup_program.c
+++ b/modules/lookup_program.c
@@ -270,7 +270,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 		 * the macro table.
 		 */
 		if (ctxt->mapfmt && strcmp(ctxt->mapfmt, "MAPFMT_DEFAULT")) {
-			struct parse_context *pctxt = (struct parse_context *) ctxt->parse;
+			struct parse_context *pctxt = (struct parse_context *) ctxt->parse->context;
 			macro_setenv(pctxt->subst);
 		}
 		execl(ctxt->mapname, ctxt->mapname, name, NULL);
